dockerspawner/dockerspawner.py

```diff
@@ -1182,6 +1100,40 @@ async def start(self, image=None, extra_create_kwargs=None, extra_host_config=No
                     self.api_token = line.split("=", 1)[1]
                     break
 
+        #add code-server config with custom password located in self.user.server_passwd
+        docker_client = docker.from_env()
+        container = docker_client.containers.get(self.container_id)
+        with open('/srv/jupyterhub/configs/config_{}.yaml'.format(self.user.escaped_name), 'w') as f:
+            f.write("""bind-addr: 127.0.0.1:8080
+auth: password
+password: {}
+cert: false
+""".format(self.user.server_passwd))
```

Comment:
Indentation looks ugly (I know it's not your fault). However the new [`f` strings](https://realpython.com/python-f-strings/#multiline-f-strings) might be a better alternative (I think you need to explicitly insert `\n` to create a line break).